modules/cfe_testcase/src/time_current_test.c

```diff
@@ -34,91 +34,81 @@
+    CFE_TIME_SysTime_t Start;
+    CFE_TIME_SysTime_t End;
     CFE_TIME_SysTime_t TAI;
     CFE_TIME_SysTime_t UTC;
     CFE_TIME_SysTime_t MET;
     CFE_TIME_SysTime_t STCF;
+    CFE_TIME_SysTime_t Range;
     uint32             METSeconds;
     uint32             METSubSeconds;
-
     int16              LeapSeconds;
     CFE_TIME_SysTime_t Buf;
 
-    char timeBuf1[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
-    char timeBuf2[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
-
-    OS_GetLocalTime(&start);
-
-    Time          = CFE_TIME_GetTime();
+    Start         = CFE_TIME_GetTime();
     TAI           = CFE_TIME_GetTAI();
     UTC           = CFE_TIME_GetUTC();
     MET           = CFE_TIME_GetMET();
     METSeconds    = CFE_TIME_GetMETseconds();
     METSubSeconds = CFE_TIME_GetMETsubsecs();
     STCF          = CFE_TIME_GetSTCF();
     LeapSeconds   = CFE_TIME_GetLeapSeconds();
+    End           = CFE_TIME_GetTime();
 
-    OS_GetLocalTime(&end);
-
-    CFE_TIME_Print(timeBuf1, Time);
-    UtPrintf("The current time is (%ld) %s", (long)Time.Seconds, timeBuf1);
-
-    difference = OS_TimeSubtract(end, start);
+    Range = CFE_TIME_Subtract(End, Start);
 
 #if (CFE_MISSION_TIME_CFG_DEFAULT_TAI == true)
-    CFE_TIME_Print(timeBuf2, TAI);
-    UtAssert_True(TimeInRange(Time, TAI, difference), "Get Time (%s) = TAI (%s)", timeBuf1, timeBuf2);
+    TimeInRange(Start, TAI, Range, "default time vs TAI");
 #else
-    CFE_TIME_Print(timeBuf2, UTC);
-    UtAssert_True(TimeInRange(Time, UTC, difference), "Get Time (%s) = UTC(%s)", timeBuf1, timeBuf2);
+    TimeInRange(Start, UTC, Range, "default time vs UTC");
 #endif
 
-    CFE_TIME_Print(timeBuf1, TAI);
     Buf = CFE_TIME_Add(MET, STCF);
-    CFE_TIME_Print(timeBuf2, Buf);
-    UtAssert_True(TimeInRange(TAI, Buf, difference), "TAI (%s) = MET + STCF (%s)", timeBuf1, timeBuf2);
+    TimeInRange(TAI, Buf, Range, "TAI vs MET + STCF");
 
-    CFE_TIME_Print(timeBuf1, UTC);
     Buf.Seconds = Buf.Seconds - LeapSeconds;
-    CFE_TIME_Print(timeBuf2, Buf);
-    UtAssert_True(TimeInRange(UTC, Buf, difference), "UTC (%s) = MET + STCF - Leap Seconds (%s)", timeBuf1, timeBuf2);
-
-    CFE_TIME_Print(timeBuf1, MET);
-    Buf.Seconds    = METSeconds;
-    Buf.Subseconds = MET.Subseconds;
-    CFE_TIME_Print(timeBuf2, Buf);
-    UtAssert_True(TimeInRange(MET, Buf, difference), "MET (%s) = METSeconds (%s)", timeBuf1, timeBuf2);
-
-    Buf.Seconds    = MET.Seconds;
-    Buf.Subseconds = METSubSeconds;
-    CFE_TIME_Print(timeBuf2, Buf);
-    UtAssert_True(TimeInRange(MET, Buf, difference), "MET (%s) = METSubSeconds (%s)", timeBuf1, timeBuf2);
+    TimeInRange(UTC, Buf, Range, "UTC vs MET + STCF - Leap Seconds");
+
+    /* Handle rollover */
+    if (METSubSeconds < MET.Subseconds)
+    {
+        UtAssert_UINT32_EQ(MET.Seconds + 1, METSeconds);
+    }
+    else
+    {
+        UtAssert_UINT32_EQ(MET.Seconds, METSeconds);
+    }
+    UtAssert_UINT32_LTEQ(METSubSeconds - MET.Subseconds, Range.Subseconds);
```

Comment:
@skliper will this work for the rollover case too?

Reply:
Yeah, just wraps since they are unsigned.